dromedary.py

```diff
@@ -163,6 +180,166 @@ def print_output(out: subprocess.CompletedProcess) -> None:
     print(std_out + err_out)
 
 
+###############################################################################
+def run_cmd_output(cmd_args: List[Any], cmd_env: Optional[Dict[str, str]]) -> str:
+    """Run the given command in a shell.
+
+    Prints the output of the process in "real time".
+
+    Args:
+        cmd_args (List[Any]): The command and it's arguments to run.
+        cmd_env (Optional[Dict[str, str]]): The environment to run the command
+                                            in.
+
+    Returns:
+        str: The accumulated stdout and stderr output of the process.
+    """
+    std_out: str = ""
+    with subprocess.Popen(
+        " ".join(map(lambda e: f"'{str(e)}'", cmd_args)),
```

Comment:
Feedback from @grievejia :

If cmd_args is a list of string, then this is just `" ".join(cmd_args)`.
Taking a step back though -- I'm not really sure how much value this big block of code brings to the table, compared to just doing a simple `subprocess.run(cmd_args)`. If we really don't like the buffered capture behavior, then it can be disabled by simple set `bufsize=1` which imo is far more straightforward than all the `poll()` that's happening here.
Also is there a strong reason why stdout and stderr are merged into a single string?

Reply:
It can never be just a `" ".join(cmd_args)`, as each argument has to be quoted, so it is at least a `" ".join(map(lambda e: f"'{e}'", cmd_args))`. Which does `str(e)` (call `e.__str__()`, even for strings). 
I'm going to add that to the function's documentation.

Setting `bufize=1` means reading one line after the other. This does not work if the command does some kind of "fancy" progress display involving e.g. backspace. Which Opam does not do as of now, so it would be safe to use until Opam changes that, if you prefer that. 
As the switch creation and package installation can take quite some time, using fully buffered output is not an option.

The reason to merge stdout and stderr is to not have to do the equivalent of a `select` on `stdout` and `stderr` in Python, as doing the "naive" version of just reading from stderr too is going to block forever, if no data is waiting in one of the pipes.

Reply:
By not parsing the name of the switch any more, there is no need for pipes, so I can use a single  `subprocess.run` without pipes, which solves all the above problems.